de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/service/BPartnerCreditLimitRepository.java

```diff
@@ -118,4 +119,18 @@ private CreditLimitType retrieveCreditLimitTypePOJO(final int C_CreditLimit_Type
 				.seqNo(type.getSeqNo())
 				.build();
 	}
+
+	public Optional<I_C_BPartner_CreditLimit> retrieveCreditLimitByBPartnerId(final int bpartnerId, final int typeId)
+	{
+		return Services.get(IQueryBL.class)
+				.createQueryBuilder(I_C_BPartner_CreditLimit.class)
+				.addEqualsFilter(I_C_BPartner_CreditLimit.COLUMNNAME_C_BPartner_ID, bpartnerId)
+				.addEqualsFilter(I_C_BPartner_CreditLimit.COLUMNNAME_Processed, true)
+				.addEqualsFilter(I_C_BPartner_CreditLimit.COLUMNNAME_C_CreditLimit_Type_ID, typeId)
+				.addOnlyActiveRecordsFilter()
+				.addOnlyContextClient()
+				.create()
+				.stream()
+				.findFirst();
```

Comment:
can there be multiple matching C_BPartner_CreditLimits?
if yes, please make sure (by ordering) that we get the right one
if no, please add a UC if non exists yet, and use findFirstOnly

Reply:
Yes, can be